internal/kaniko/execute.go

```diff
@@ -105,6 +113,56 @@ func (k *Config) processRegistryMirrors() []string {
 	return strings.Split(k.RegistryMirrors, ",")
 }
 
+func (k *Config) processRegistryMaps() (string, error) {
+	if len(k.RegistryMaps) == 0 {
+		return registryMapsInConfig()
+	}
+
+	predefinedMaps, err := registryMapsInConfig()
+	if err != nil {
+		return "", err
+	}
+
+	if predefinedMaps == "" {
+		return k.RegistryMaps, nil
+	}
+
+	return k.RegistryMaps + ";" + predefinedMaps, nil
+}
+
+func registryMapsInConfig() (string, error) {
+	regConfig := os.Getenv("REDHAT_REGISTRY_CONFIG")
+	if regConfig == "" {
+		return "", nil
+	}
+
+	if _, err := os.Stat(regConfig); err != nil && os.IsNotExist(err) {
+		return "", nil
+	}
+
+	// Parse the Red-Hat registry config file.
+	f, err := os.ReadFile(regConfig)
+	if err != nil {
+		return "", fmt.Errorf("failed to open registry config file: %w", err)
+	}
+
+	var registries = sysregistriesv2.V2RegistriesConf{}
```

Comment:
Why integrate the kaniko Action to the RedHat registry configuration file? I'd rather integrate it with the CloudBees registries configuration file only, no?
(Because of this kaniko Action I am thinking of removing the RedHat `registries.conf` file again for now (that we just introduced yesterday) because otherwise it ends up being baked into every image kaniko builds, unless the kaniko Action explicitly ignores it which however may not be desired in some edgy cases either where customers would want to add a custom `registries.conf` file to their image explicitly.)

Reply:
> I'd rather integrate it with the CloudBees registries configuration file only, no?

Yeah, i did that in my first iteration but the build would not run in the workflow because to parse the CloudBees registry configuration file, I would need to import the private repo `calculi-corp/registry-config` which would not work in our build system since the `kaniko` repo is in a different GitHub org from the `registry-config` repo, and moreover this would also be a problem for external users forking this repo to make changes in the future.

Reply:
Okay, but the fix to the registries.json will be promoted later, right? Thus, you'll be unblocked on that end tomorrow.

Reply:
It's not related to the fix, this workflow was tested in the Staging env where the fix was already promoted. The problem is un-marshalling the JSON into the structure defined in the `calculi-corp/registry-config` (private repo), which needs to be imported here to un-marshal the JSON and the issues I ran into doing that, which is mentioned in my earlier comment. Unless you are thinking processing the JSON as a generic `map[string]interface{}`, which we could technically do, but I want to make sure that is what you are expecting here?

Reply:
ah, I see. hm, instead of not leveraging the Go structs we have defined for the `registries.json` configuration file I am inclined to make the registry-config repo public. WDYT?

Reply:
`registry-config` repo was made [Public](https://cloudbees.atlassian.net/browse/CBP-3954) and the code updated correspondingly.